src/components/splashScreen/TitleBadge.tsx

```diff
@@ -1,14 +1,36 @@
+import { useState } from "react";
 import styles from "./TitleBadge.module.css";
 
 type TitleBadgeProps = {
   isPaniniOrdered: boolean;
+  handleTransition: () => void;
 };
 
 export default function TitleBadge(props: TitleBadgeProps) {
+  const [isButtonAvailable, setIsButtonAvailable] = useState(true);
+  const handleClick = () => {
+    setIsButtonAvailable(false);
+    setTimeout(() => {
+      setIsButtonAvailable(true);
+    }, 4000);
+  };
+  const buttonStyle = {
```

Comment:
**nitpick:** It's considered a bad practice to put styles object inside component body, as this object reference has to be recreated on every render, it leads to worse app performance (of course when it's overused). You can fix it by putting this object outside component body, so it's created only once